### ISSUE TITLE
Poll Password

### ISSUE THREAD

**Issue by PrinxeShamar** [PollBuddy/PollBuddy#728]
**Is your feature request related to a problem? Please describe.**
Currently once polls are open users can begin to access it. This poses a problem that we may want users to have a passcode to access this poll. Example: The RCOS attendance poll is open from 4 to 6. However users can open the poll and answer it. Because different rooms will have different exam times making the poll due at say 5 isn't the best solution.

**Describe the solution you'd like**
We can introduce a poll password. This password can be displayed once the creator is ready. Example: In Small groups, Mentors can present the code once the class is ready to leave.

Poll Access Overview
Once trying to access a poll, a page with a input pops up that accepts text. Once the user enters in the password a request is sent to the backend and is validated against a 5 digit randomly generated passcode. If the check pass continue with the poll request and then actually display the poll.

Poll Password Display
To make the job easy for displaying a poll password, there should be button that makes a pop up that displays the code so everything can be done in app.

**Comment by hbh7** [comment]
Good idea! IMO, we should probably make configurable poll availability, perhaps thru a series of edit options where you can pick options like "open always", "open during a range of time", and "open with password", or something along those lines.

**Comment by PrinxeShamar** [comment]
I feel like there can be some use cases for a time and a poll password. For example RCOS leadership wants to set bounds for first accepting results and latest results example no attendance collected before 4:30 (atleast 30 mins into class) and no attendance after 6:00pm (last time to get attendance). Also it will serve a benefit as time controls whether it get displayed in the My Polls section. If it was always open all polls would show which can make for not a nice user experience.

**Comment by PrinxeShamar** [comment]
[Poll Password](https://docs.google.com/document/d/1jXYlbJqfHTNYZyaCgjwbNLBWcGk1cHFNZPCnRsrh-p0/edit#) Here is a design for the Poll Password feature. It should point you in all the right places and help with the flow. @xDiggy

**Comment by hbh7** [comment]
Yeah I would agree. It would depend on the poll's use case what would make sense, so having it be easily selectable which option(s) to use would be nice. Might also good to have a toggle for "Require all of these" vs "Require any of these", such as having the user be within a time range AND have a password, or be within a time range OR have a password. Your doc looks very nice btw! I made a couple of small comments for clarity, just so you know.
